tic-tac-toe/spec/grid_spec.rb

```diff
@@ -0,0 +1,232 @@
+require_relative "../lib/grid.rb"
+require_relative "../lib/game.rb"
+require_relative "spec_helper.rb"
+
+RSpec.describe Grid do
+  subject { Grid.new }
+  let (:player1) { Player.new({name: "player1", symb: "x"})}
+  let (:player2) { Player.new({name: "player2", symb: "o"})}
+
+  describe '#new' do
+    it 'init' do
+      result = Grid.new
+      expect(result).not_to be_nil
+    end
+  end
+  describe '#put_piece' do
+    it 'puts mark ' do
+      grid = Grid.new
+      Array.any_instance.stub(:shuffle){ [player2, player1] }
+      game = Game.new( [player1, player2] )
+
+      dbl=double("Put_piece", :index=>3, :symb =>'o')
+      result1=grid.put_piece(dbl.index, dbl.symb)
+
+      expect(game.current_player_symb).to eq(dbl.symb)
+    end
+  end
+  describe '#finished?' do
+    it 'returns true if all marks on the main diagonal  are (X)s' do
+      grid = Grid.new
+      dbl1=double("Put_piece", :index=>1, :symb =>'x')
+      result1=grid.put_piece(dbl1.index.to_i - 1, dbl1.symb)
+
+      dbl2=double("Put_piece", :index=>5, :symb =>'x')
+      result2=grid.put_piece(dbl2.index.to_i - 1, dbl2.symb)
+
+      dbl3=double("Put_piece", :index=>9, :symb =>'x')
+      result3=grid.put_piece(dbl3.index.to_i - 1, dbl3.symb)
+
+      result = grid.finished?
+      expect(result).to be true
+    end
+
+    it 'returns true if all marks on the main diagonal are (O)s' do
+      grid = Grid.new
+      dbl1=double("Put_piece", :index=>1, :symb =>'o')
+      result1=grid.put_piece(dbl1.index.to_i - 1, dbl1.symb)
+
+      dbl2=double("Put_piece", :index=>5, :symb =>'o')
+      result2=grid.put_piece(dbl2.index.to_i - 1, dbl2.symb)
+
+      dbl3=double("Put_piece", :index=>9, :symb =>'o')
+      result3=grid.put_piece(dbl3.index.to_i - 1, dbl3.symb)
+
+      result = grid.finished?
+      expect(result).to be true
+    end
+
+    it 'returns true if all marks on the secondary diagonal  are (X)s' do
+      grid = Grid.new
+      dbl1=double("Put_piece", :index=>3, :symb =>'x')
+      result1=grid.put_piece(dbl1.index.to_i - 1, dbl1.symb)
+
+      dbl2=double("Put_piece", :index=>5, :symb =>'x')
+      result2=grid.put_piece(dbl2.index.to_i - 1, dbl2.symb)
+
+      dbl3=double("Put_piece", :index=>7, :symb =>'x')
+      result3=grid.put_piece(dbl3.index.to_i - 1, dbl3.symb)
+
+      result = grid.finished?
+      expect(result).to be true
+    end
+
+    it 'returns true if all marks on the secondary diagonal are (O)s' do
```

Comment:
What about when you expect it to return `false`?

It is best practice to cover both negative and positive scenarios for each method. Otherwise, how does the person running test know if you method actually works the way you expect it to?